R/dataframe-s3-base.R

```diff
@@ -324,11 +341,18 @@ tail.polars_data_frame <- function(x, n = 6L, ...) x$tail(n = n)
               if (sign_start == 1) "negative" else "positive",
               loc
             )
-          )
+          ),
+          call = error_env
         )
       }
       cols[j]
     } else if (is_bare_character(j)) {
+      if (!all(j %in% cols)) {
+        abort(
+          sprintf("Column(s) not found: %s.", oxford_comma(setdiff(j, cols))),
+          call = error_env
+        )
+      }
```

Comment:
When `x` was only a DataFrame, we didn't need this because `$select()` at the end would give a nice error message. But since `x` can now be a LazyFrame, `$select()` doesn't resolve the schema and doesn't error, so we need to "manually error" early.

Reply:
In my opinion it is needed for DataFrame too.
Since `$select()` accepts special strings like `"*"`, I think this check is necessary even if it is a DataFrame.

I will do that in #343

Reply:
This change applies to both DataFrame and LazyFrame in the current PR. I don't think you need to add it in #343.